File: api_control_neumaticos/Models/Movil.cs
using System;
using System.Collections.Generic;

namespace api_control_neumaticos.Models
{
    public partial class Movil
    {
        public int IdMovil { get; set; }
        public required string Patente { get; set; }
        public required string Marca { get; set; }
        public required string  Modelo { get; set; }
        public int Ejes { get; set; }
        public int TipoMovil { get; set; }
        
        // Nueva propiedad para la relación con Bodega
        public int? ID_BODEGA { get; set; }
        public virtual required Bodega Bodega { get; set; }

        // Nueva propiedad para CantidadNeumaticos
        public int? CantidadNeumaticos { get; set; }

        // Nueva propiedad para Estado
        public int Estado { get; set; }

        public DateTime? FechaUltimaComprobacion { get; set; }
        public ICollection<HistorialMovil> Historiales { get; set; } = new List<HistorialMovil>();
    }
}

File: flutter_control_neumaticos/lib/models/admin/movil.dart
class Movil {
  final String patente;
  final String marca;
  final String modelo;
  final int ejes;
  final int cantidadNeumaticos;
  final int tipoMovil;
  final int estado;
  final int bodega;

  Movil({
    required this.patente,
    required this.marca,
    required this.modelo,
    required this.ejes,
    required this.cantidadNeumaticos,
    required this.tipoMovil,
    required this.estado,
    required this.bodega,
  });

  Map<String, dynamic> toJson() {
    return {
      'PATENTE': patente,
      'MARCA': marca,
      'MODELO': modelo,
      'EJES': ejes,
      'TIPO_MOVIL': tipoMovil,
      'ID_BODEGA': bodega,
      'CANTIDAD_NEUMATICOS': cantidadNeumaticos,
      'ESTADO': estado,
    };
  }
  factory Movil.fromJson(Map<String, dynamic> json) {
    return Movil(
      patente: json['PATENTE'],
      marca: json['MARCA'],
      modelo: json['MODELO'],
      ejes: json['EJES'],
      cantidadNeumaticos: json['CANTIDAD_NEUMATICOS'],
      tipoMovil: json['TIPO_MOVIL'],
      estado: json['ESTADO'],
      bodega: json['ID_BODEGA'],
    );
  }
}
